trace-dispatcher/src/Cardano/Logging/DocuGenerator.hs

```diff
@@ -55,65 +52,58 @@ unpackDocu (DocuTracer b)    = b
 unpackDocu (DocuMetric b)    = b
 unpackDocu (DocuDatapoint b) = b
 
-anyProto :: a
-anyProto = undefined
+
+-- | Calls the tracers in a documetation control mode,
+-- and returns a DocCollector, from which the documentation gets generated
+documentTracers :: Documented a -> [Trace IO a] -> IO DocCollector
+documentTracers (Documented documented) tracers = do
+    let docIdx = zip documented [0..]
+    coll <- fmap DocCollector (liftIO $ newIORef Map.empty)
+    mapM_ (docTrace docIdx coll) tracers
+    pure coll
+  where
+    docTrace docIdx dc (Trace tr) =
+      mapM_
+        (\ (DocMsg {..}, idx) ->
+            T.traceWith tr (emptyLoggingContext {lcNamespace = dmNamespace},
+                            Left (Document idx dmMarkdown dmMetricsMD dc)))
+        docIdx
 
 showT :: Show a => a -> Text
 showT = pack . show
 
-appDoc :: (a -> b) -> DocMsg a -> DocMsg b
-appDoc f DocMsg {..} = DocMsg  (f dmPrototype) dmMetricsMD dmMarkdown
+-- appDoc :: (Namespace -> Namespace) -> DocMsg a -> DocMsg b
+-- appDoc f DocMsg {..} = DocMsg  (f dmNamespace) dmMetricsMD dmMarkdown
+--
+-- mapDoc :: (Namespace -> Namespace) -> [DocMsg a] -> [DocMsg b]
+-- mapDoc f = map (appDoc f)
```

Comment:
Why are these commented out?

Reply:
Removed them, cause I use the interface on Documented level, instead of DocMsg